product/integration/tests-integration/tests/src/test/java/org/wso2/das4is/integration/tests/is/ISAlertLongSessionsAnalyticsTestCase.java

```diff
@@ -139,18 +139,33 @@ public void publishData() throws Exception {
 
         //run spark scripts for session duration calculations
         analyticsStub.executeScriptInBackground("IsAnalytics-SparkScript-SessionManagement");
-        Thread.sleep(15000);
+        Thread.sleep(30000);
         analyticsStub.executeScriptInBackground("ISAnalytics-SparkScript-AlertLongSessions");
-        //wait until execution plan is triggered
-        Thread.sleep(180000);
     }
 
     @Test(groups = "wso2.analytics.is", description = "Checking total long session count", dependsOnMethods = "publishData")
     public void retrieveTableCountTest() throws AnalyticsServiceException, AnalyticsException {
-        long eventCount = analyticsDataAPI
-                .getRecordCount(MultitenantConstants.SUPER_TENANT_ID, "ORG_WSO2_IS_ANALYTICS_STREAM_LONGSESSIONS",
-                        Long.MIN_VALUE, Long.MAX_VALUE);
-        Assert.assertEquals(eventCount, 2, "========== Total long session count is invalid ==========");
+
+        final int EXPECTED_COUNT = 2;
+        final int MAX_WAIT_COUNT = 5;
+        final String LONGSESSIONS_TABLE = "ORG_WSO2_IS_ANALYTICS_STREAM_LONGSESSIONS";
+        int waitCount = 0;
+
+        while (analyticsDataAPI.getRecordCount(MultitenantConstants.SUPER_TENANT_ID, LONGSESSIONS_TABLE,
+                Long.MIN_VALUE, Long.MAX_VALUE) < EXPECTED_COUNT && waitCount < MAX_WAIT_COUNT) {
+
```

Comment:
Fixed with https://github.com/wso2/analytics-is/pull/288/commits/06102bc1ec294d9983507efcfe0b24212cebb984